_tutorials/development.md

```diff
@@ -22,32 +25,67 @@ $ ssh <NETID>@{{site.data.constants.semester }}-{{ site.data.constants.departmen
 
-### Enabling password-less authentication with your VM
+### Configuring Git
+
+**Once you are in your VM**, you'll need to set up some global defaults. The name and email you input here will be used to mark your commits.
 
-If you have generated an SSH key on your local machine (see [Using SSH keys](#using-ssh-keys) below), you can export your public key to your VM using the following command:
 ```console
-$ ssh-copy-id <NETID>@{{site.data.constants.semester }}-{{ site.data.constants.department_code }}{{ site.data.constants.course_number }}-<NUM>.cs.illinois.edu
+git config --global user.name "FIRST_NAME LAST_NAME"
+git config --global user.email NETID@example.com
 ```
 
-## Authenticating with GitHub
+Make sure to replace `FIRST_NAME` and `LAST_NAME` and `NETID@example.com` with your information. For example:
+
+```console
+git config --global user.name "Lawrence Angrave"
+git config --global user.email "angrave@illinois.edu"
+```
 
-There are two ways that we recommend for authenticating with GitHub
+### Cloning your course repository
 
-### Using SSH keys
+Checkout your repository as follows:
+
+```console
+git clone https://github.com/illinois-cs-coursework/{{site.data.constants.semester }}_{{ site.data.constants.department_code }}{{site.data.constants.course_number}}_NETID.git
+```
+
+Unfortunately, cloning a repository is not so easy. If you follow git's prompt and submit your username and password, you will likely see an error message like the following. If so, continue to the instructions in [Authenticating with GitHub](#authenticating-with-github).
+
+```console
+remote: Support for password authentication was removed on August 13, 2021.
+remote: Please see https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls for information on currently recommended modes of authentication.
+fatal: Authentication failed for 'https://github.com/illinois-cs-coursework/{{site.data.constants.semester }}_{{ site.data.constants.department_code }}{{site.data.constants.course_number}}_NETID.git'
+```
+
+### Authenticating with GitHub
+
+There are three ways that we recommend for authenticating with GitHub
+
+#### Option 1: Using SSH keys
 
 GitHub provides an easy way to add SSH keys to your account. GitHub's documentation can be found [here](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account?platform=linux). This method has the added benefit that you can reuse the same SSH key for password-less authentication to your CS341 VM.
 
 In summary:
 
-1. Create an SSH key on your local machine if you do not already have one. Currently, we recommend the EdDSA key algorithm. You can create an EdDSA key with the command `ssh-keygen -t ed25519 -a 100 -f`.
```

Comment:
"This command generates a private and public key for authentication. It will prompt you to enter a name for the file  storing your private key, and will generate a .pub file for your public key." 

maybe?

Reply:
That's a good idea, it isn't clear exactly what the command does. Updated.